inst/include/mode/solver.hpp

```diff
@@ -99,13 +99,38 @@ class solver {
     }
   }
 
-  void reset(double t, std::vector<double> y) {
-    stepper_.reset(t, y);
+  void reset(double t,
```

Comment:
Could do away with this function as it's essentially the same as `set_state`, but would need to pass an iterator instead of a vector https://github.com/mrc-ide/mode/pull/8/files#diff-8c2c00c231ed8b1864f07acd80fcd795bec1b09543039b19c84bc3d6de9b3bf5R79 and https://github.com/mrc-ide/mode/pull/8/files#diff-6df4c22acbd9e0958ea70ddc605ef1d2b7369270ccea7732e6d3a24de77788a1R29

Reply:
GitHub is making those links hard to follow (I think they don't resolve unless my diff settings are the same as yours).  But if iterator vs vector is the only difference you could overload `set_state`:

```
  void set_state(double t,
                 const std::vector<double>& state,
                 bool reset_step_size) {
    set_state(t, state.begin(), reset_step_size);
  }
```

(retaining the existing definition for `solver::set_state`) and you can cut out the duplication